publisher/publisher.go

```diff
@@ -208,10 +213,23 @@ func (p *Publisher) Publish(ctx context.Context, msg interface{}, opts ...Option
 
 func ensureTopic(ctx context.Context, name string, namespace *servicebus.Namespace, opts ...servicebus.TopicManagementOption) (*servicebus.TopicEntity, error) {
 	tm := namespace.NewTopicManager()
-	te, err := tm.Get(ctx, name)
-	if err == nil {
+	ensure := func() (interface{}, error) {
+		te, err := tm.Get(ctx, name)
+		if err == nil {
+			return te, nil
+		}
+		te, err = tm.Put(ctx, name, opts...)
+		if err != nil {
+			// let all errors be retryable for now. application only hit this once on topic creation.
```

Comment:
Even 429's? Vague preferenc to just retry >500 and 409 but you retry is slow enough it's not a big deal.

Reply:
Question is how easy is the status code to pull out of the err. If it's string matching then agreee retrying all might be better.

Reply:
easy, yes, easy.
reliable... well, I depend on the implementation of the azure-serbice-bus-go.
granted, we own it, and any change in that package has to be accounted for in go-shuttle, so probably very low risk.
just extra complexity that I don't think will bring value in practice.